src/XAMLTools.Core/XAMLCombine/XAMLCombiner.cs

```diff
@@ -149,43 +161,99 @@ public string Combine(IEnumerable<string> sourceFiles, string targetFile)
-                        {
-                            key = importedElement.Attributes["Key"].Value;
-                        }
-                        else if (importedElement.HasAttribute("x:Key"))
-                        {
-                            key = importedElement.Attributes["x:Key"].Value;
-                        }
-                        else if (importedElement.HasAttribute("TargetType"))
+                    // Merged resource dictionaries (at the top)
+                    if (node.Name == MergedDictionariesString)
+                    {
+                        if (importMergedResourceDictionariesReferences)
                         {
-                            key = importedElement.Attributes["TargetType"].Value;
-                        }
+                            if (rootNode.ChildNodes.Count == 0)
+                            {
+                                mergedDictionariesListNode = finalDocument.CreateElement(MergedDictionariesString, "http://schemas.microsoft.com/winfx/2006/xaml/presentation");
+                                rootNode.AppendChild(mergedDictionariesListNode);
+                            }
+                            else if (rootNode.FirstChild.Name != MergedDictionariesString)
+                            {
+                                mergedDictionariesListNode = finalDocument.CreateElement(MergedDictionariesString, "http://schemas.microsoft.com/winfx/2006/xaml/presentation");
+                                rootNode.InsertBefore(mergedDictionariesListNode, rootNode.FirstChild);
+                            }
 
-                        if (string.IsNullOrEmpty(key) == false)
-                        {
-                            // Check key unique
-                            if (keys.Contains(key))
+                            if (mergedDictionariesListNode == null)
                             {
                                 continue;
                             }
 
-                            keys.Add(key);
+                            var currentMergedSources = mergedDictionariesListNode.ChildNodes.OfType<XmlElement>()
+                                .Select(node => node.GetAttribute("Source"))
+                                .Where(source => !string.IsNullOrEmpty(source)).ToList();
+
+                            foreach (var mergedDictionaryReference in xmlElement.ChildNodes)
+                            {
+                                if (mergedDictionaryReference is not XmlElement mergedDictionaryReferenceElement || mergedDictionaryReferenceElement.Name != "ResourceDictionary")
+                                {
+                                    continue;
+                                }
+
+                                var sourceValue = mergedDictionaryReferenceElement.GetAttribute("Source");
+                                if (string.IsNullOrEmpty(sourceValue))
+                                {
+                                    continue;
```

Comment:
Do we want to consider merging values from merged dictionaries here? Maybe we should log a warning because a merged dictionary *without* source will probably contain resources and will be ignored.